modules/quadbin/redshift/lib/tilecover/__init__.py

```diff
@@ -0,0 +1,218 @@
+# Copyright (c) 2014, Morgan Herlocker (JavaScript implementation)
+# Copyright (c) 2021, CARTO
+
+from __future__ import division
+import math
+
+d2r = math.pi / 180
+
+
+def get_tiles(geom, limits):
+    import mercantile
+
+    i = None
+    tile = None
+    coords = geom['coordinates']
+    min_zoom = limits['min_zoom']
+    max_zoom = limits['max_zoom']
+    tile_hash = {}
+    tiles = []
+
+    if geom['type'] == 'Point':
+        return [mercantile.tile(coords[0], coords[1], max_zoom)]
+    elif geom['type'] == 'MultiPoint':
+        for i in range(0, len(coords)):
+            tile = mercantile.tile(coords[i][0], coords[i][1], max_zoom)
+            tile_hash[to_id(tile[0], tile[1], tile[2])] = True
+    elif geom['type'] == 'LineString':
+        line_cover(tile_hash, coords, max_zoom, None)
+    elif geom['type'] == 'MultiLineString':
+        for coord in coords:
+            line_cover(tile_hash, coord, max_zoom, None)
+    elif geom['type'] == 'Polygon':
+        polygon_cover(tile_hash, tiles, coords, max_zoom)
+    elif geom['type'] == 'MultiPolygon':
+        for coord in coords:
+            polygon_cover(tile_hash, tiles, coord, max_zoom)
+    else:
+        raise Exception('Geometry type not implemented')
+
+    if min_zoom != max_zoom:
+        #  sync tile hash and tile array so that both contain the same tiles
+        tiles_length = len(tiles)
+        append_hash_tiles(tile_hash, tiles)
+        for i in range(0, tiles_length):
+            t = tiles[i]
+            tile_hash[to_id(t[0], t[1], t[2])] = True
+
+        return merge_tiles(tile_hash, tiles, limits)
+
+    append_hash_tiles(tile_hash, tiles)
+    return tiles
+
+
+def merge_tiles(tile_hash, tiles, limits):
```

Comment:
So the RS polyfills compacts the tiles?
If so shouldn't that be optional? (and have by default the same behaviour we have in the other providers)

Reply:
Has we have discussed. It looks like we are not entering that merge_tiles function because of setting both min and max zoom as `resolution`.